### PR TITLE
refactor: convert ECR image of RuntimeConfig into `map` in stack package

### PR DESCRIPTION
**Pull request by KollaAdithya**
With This PR we convert the current stack package  
https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/deploy/cloudformation/stack/workload.go#L69

into

```yaml
type RuntimeConfig struct {
   Images map[string]ECRImage // Optional. Image location in an ECR repository.
}
```
In this way we can use same struct `ECRImage` for both main and sidecar containers.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

**Review by iamhopaul123 — APPROVED** [review]

File: internal/pkg/cli/deploy/workload.go [hunk 1]
```diff
@@ -93,10 +93,7 @@ type fileReader interface {
 
 // StackRuntimeConfiguration contains runtime configuration for a workload CloudFormation stack.
 type StackRuntimeConfiguration struct {
-	// Use *string for three states (see https://github.com/aws/copilot-cli/pull/3268#discussion_r806060230)
-	// This is mainly to keep the `workload package` behavior backward-compatible, otherwise our old pipeline buildspec would break,
-	// since previously we parsed the env region from a mock ECR URL that we generated from `workload package``.
-	ImageDigest        *string
+	ImageDigests       map[string]ContainerImageIdentifier
```
Comment (efekarakus) [inline internal/pkg/cli/deploy/workload.go]: nit:
```suggestion
	ImageDigests       map[string /* container name */]ContainerImageIdentifier // Samesies elsewhere where a field is defined.
```

File: internal/pkg/cli/deploy/workload.go [hunk 2]
```diff
@@ -336,7 +333,7 @@ func (img ContainerImageIdentifier) customOrGitTag() string {
 	return img.GitShortCommitTag
 }
 
-func (d *workloadDeployer) uploadContainerImage(imgBuilderPusher imageBuilderPusher) (*string, error) {
+func (d *workloadDeployer) uploadContainerImage(imgBuilderPusher imageBuilderPusher) (map[string]ContainerImageIdentifier, error) {
```
Comment (efekarakus) [inline internal/pkg/cli/deploy/workload.go]: nit:
```suggestion
func (d *workloadDeployer) uploadContainerImages(imgBuilderPusher imageBuilderPusher) (map[string]ContainerImageIdentifier, error) {
```

File: internal/pkg/cli/deploy/workload.go [hunk 3]
```diff
@@ -557,15 +553,19 @@ func (d *workloadDeployer) runtimeConfig(in *StackRuntimeConfiguration) (*stack.
 			EnvVersion:               envVersion,
 		}, nil
 	}
-	return &stack.RuntimeConfig{
-		AddonsTemplateURL: in.AddonsURL,
-		EnvFileARN:        in.EnvFileARN,
-		AdditionalTags:    in.Tags,
-		Image: &stack.ECRImage{
+	images := make(map[string]stack.ECRImage)
```
Comment (efekarakus) [inline internal/pkg/cli/deploy/workload.go]: nit:
```suggestion
	images := make(map[string]stack.ECRImage, len(in.ImageDigests))
```

File: internal/pkg/deploy/cloudformation/stack/backend_svc_test.go [hunk 4]
```diff
@@ -273,9 +273,11 @@ Outputs:
 			},
 			Manifest: mft,
 			RuntimeConfig: RuntimeConfig{
-				Image: &ECRImage{
-					RepoURL:  testImageRepoURL,
-					ImageTag: testImageTag,
+				Images: map[string]ECRImage{
```
Comment (efekarakus) [inline internal/pkg/deploy/cloudformation/stack/backend_svc_test.go:276]: Do we have any test cases with more than one entry in the map? to make sure we are translating multiple container images

**Review by efekarakus — COMMENTED** [review]
Looks great!

File: internal/pkg/deploy/cloudformation/stack/backend_svc_test.go (hunk 4 above)
Comment (KollaAdithya) [reply internal/pkg/deploy/cloudformation/stack/backend_svc_test.go:276]: As of now we are passing only the main container config. In the upcoming PR I am adding sidecars as well, in that case I will update the testcases. Does that sound good?

Comment (efekarakus) [reply internal/pkg/deploy/cloudformation/stack/backend_svc_test.go:276]: sounds great!

**Review by efekarakus — APPROVED** [review]

**Pull request merged**